lua/customroles/hermit.lua

```diff
@@ -450,7 +450,81 @@ if CLIENT then
         return roleString .. "_jester"
     end)
 
-    -- TODO: Hermit tutorial
+
+    --------------
+    -- TUTORIAL --
+    --------------
+
+    local function GetRevealModeInfo(roleColor, revealMode, teamName, teamColor)
+        local modeString = "When joining the <span style='color: rgb(" .. teamColor.r .. ", " .. teamColor.g .. ", " .. teamColor.b .. ")'>" .. string.lower(teamName) .. "</span> team, the <span style='color: rgb(" .. roleColor.r .. ", " .. roleColor.g .. ", " .. roleColor.b .. ")'>" .. ROLE_STRINGS[ROLE_HERMIT] .. "</span>'s new role will be revealed to "
+        local revealed = true
+        if revealMode == BEGGAR_REVEAL_ALL then
+            modeString = modeString .. "everyone"
+        elseif revealMode == BEGGAR_REVEAL_TRAITORS then
+            local revealColor = ROLE_COLORS[ROLE_TRAITOR]
+            modeString = modeString .. "only <span style='color: rgb(" .. revealColor.r .. ", " .. revealColor.g .. ", " .. revealColor.b .. ")'>" .. string.lower(LANG.GetTranslation("traitors")) .. "</span>"
+        elseif revealMode == BEGGAR_REVEAL_INNOCENTS then
+            local revealColor = ROLE_COLORS[ROLE_TRAITOR]
+            modeString = modeString .. "only <span style='color: rgb(" .. revealColor.r .. ", " .. revealColor.g .. ", " .. revealColor.b .. ")'>" .. string.lower(LANG.GetTranslation("innocents")) .. "</span>"
+        elseif revealMode == BEGGAR_REVEAL_ROLES_THAT_CAN_SEE_JESTER then
+            modeString = modeString .. "any role that can see <span style='color: rgb(" .. roleColor.r .. ", " .. roleColor.g .. ", " .. roleColor.b .. ")'>" .. string.lower(LANG.GetTranslation("jesters")) .. "</span>"
+        else
+            modeString = modeString .. "nobody"
+            revealed = false
+        end
+        return modeString .. ".", revealed
+    end
+
+    AddHook("TTTTutorialRoleText", "Hermit_TTTTutorialRoleText", function(role, titleLabel)
+        if role ~= ROLE_HERMIT then return end
+
+        local roleTeam = ROLE_TEAM_JESTER
+        if hermit_is_independent:GetBool() then
```

Comment:
`player.GetRoleTeam(ROLE_HERMIT)`

Reply:
Can't do that here because the hermit might be an innocent or traitor if the keep begging convar is enabled

Reply:
Ah yea, good call